.github/workflows/test.yml

```diff
@@ -561,5 +559,185 @@ jobs:
       - uses: codecov/codecov-action@v2
         with:
           files: ./packages/generator-helper/src/__tests__/coverage/clover.xml
-          flags: generator-helper,${{ matrix.os }}
+          flags: generator-helper,${{ matrix.os }},library,binary
+          name: generator-helper-${{ matrix.os }}
+
+  #
+  # Run all tests on macOS and Windows.
+  #
+  # Unlike the other jobs, this job doesn't use Docker (and thus skips some
+  # tests that require dependencies not easily installable without Docker —
+  # specifically, SQL Server), and runs tests on Node.js 12, 14 and 16 (Linux
+  # jobs only run on Node.js 12 and 16 because Node.js 14 is tested in
```

Comment:
Is running another Node version such high cost that this is worth the difference?
If not, maybe just run all the tests on all node versions.

Reply:
Kind of. Mostly due to the number of jobs themselves and the time spent waiting for a free runner.

Reply:
On the other hand, we may try merging our Linux jobs into a single one, similar to `no-docker`. Running the setup script takes more time than running tests themselves in all packages except `client` anyway.

Reply:
Also heads up that this comment is not fully true since we decided to only run Windows and macOS tests with Node.js 16, at least for now: https://github.com/prisma/prisma/pull/10526#discussion_r770799470. It would make sense to expand the matrix if we see that the CI doesn't take too much time though.